src/backend/app/index-post/routes.js

```diff
@@ -0,0 +1,91 @@
+const db = require("../../core/database/models");
+const { StatusCodes } = require("http-status-codes");
+const jwt = require("jsonwebtoken");
+const { updatePostIndexStatus, datasourceIndexStatus, postIndexStatus, blacklistPostIndex, indexPosts } = require("../data-source/repository-post");
+
+const PostIndexHistory = db.sequelize.models.postIndexHistory;
+
+const configure = (expressApp) => {
+    expressApp.post("/index/report", async (req, res) => {
+        const indexHistory = {
+            indexer_id: req.body.indexer_id,
+            post_id: req.body.post_id,
+            status_code: req.body.status_code,
+            status: req.body.status
+        }
+        try {
+            await PostIndexHistory.create(indexHistory)
+            await updatePostIndexStatus(indexHistory)
+        } catch (err) {
+            console.log("Error : Could not create Index History");
+            throw err;
+        }
+        res
+            .status(StatusCodes.OK)
+            .send({});
+    });
+
+    expressApp.post("/index", async (req, res) => {
+        try {
+            const accessToken = jwt.sign(req.user, process.env.ACCESS_TOKEN_SECRET)
```

Comment:
@gpstrange I noticed that you are extracting the accesstoken here and passing it in the auth header to feluda's /index endpoint.

You don't need to do that. Firstly, feluda's API endpoint is not accessible to public internet. Its internal to our infrastructure. So only other services in our infrastructure can make an API call to it. in this case only kosh. So feluda assumes that only trusted services are accessing it and has no authentication layer. 

since /index endpoint will only be accessible to kosh admin, we can safely pass that request to feluda without any need to pass on the token.